engine/src/main/java/org/hibernate/validator/internal/engine/messageinterpolation/el/VariablesELContext.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine.messageinterpolation.el;
+
+import javax.el.ArrayELResolver;
+import javax.el.CompositeELResolver;
+import javax.el.ELResolver;
+import javax.el.ExpressionFactory;
+import javax.el.ListELResolver;
+import javax.el.MapELResolver;
+import javax.el.ResourceBundleELResolver;
+import javax.el.StandardELContext;
+
+/**
+ * @author Guillaume Smet
+ */
+public class VariablesELContext extends StandardELContext {
+
+	private static final ELResolver DEFAULT_RESOLVER = new CompositeELResolver() {
+
+		{
+			add( new RootResolver() );
+			add( new ArrayELResolver( true ) );
+			add( new ListELResolver( true ) );
+			add( new MapELResolver( true ) );
+			add( new ResourceBundleELResolver() );
+			add( new NoOpElResolver() );
```

Comment:
Maybe a comment to explain why this is necessary? I'm sure there's a reason, but it seems weird.